yt_dlp/extractor/huya.py

```diff
@@ -0,0 +1,140 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import hashlib
+import random
+
+from ..compat import compat_urlparse, compat_b64decode
+
+from ..utils import (
+    ExtractorError,
+    int_or_none,
+    js_to_json,
+    str_or_none,
+    try_get,
+    unescapeHTML,
+    update_url_query,
+)
+
+from .common import InfoExtractor
+
+
+class HuyaLiveIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.|m\.)?huya\.com/(?P<id>[^/#?&]+)(?:\D|$)'
+    IE_NAME = 'huya:live'
+    IE_DESC = 'huya.com'
+    TESTS = [{
+        'url': 'https://www.huya.com/572329',
+        'info_dict': {
+            'id': '572329',
+            'title': str,
+            'description': str,
+            'is_live': True,
+            'view_count': int,
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }, {
+        'url': 'https://www.huya.com/xiaoyugame',
+        'only_matching': True
+    }]
+
+    _RESOLUTION = {
+        '蓝光4M': {
+            'width': 1920,
+            'height': 1080,
+        },
+        '超清': {
+            'width': 1280,
+            'height': 720,
+        },
+        '流畅': {
+            'width': 800,
+            'height': 480
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id=video_id)
+        json_stream = self._search_regex(r'"stream":\s+"([a-zA-Z0-9+=/]+)"', webpage, 'stream', default=None)
+        if not json_stream:
+            raise ExtractorError('Video is offline', expected=True)
+        stream_data = self._parse_json(compat_b64decode(json_stream).decode(), video_id=video_id,
+                                       transform_source=js_to_json)
+        room_info = try_get(stream_data, lambda x: x['data'][0]['gameLiveInfo'])
+        if not room_info:
+            raise ExtractorError('Can not extract the room info', expected=True)
+        title = room_info.get('roomName') or room_info.get('introduction') or self._html_search_regex(
+            r'<title>([^<]+)</title>', webpage, 'title')
+        screen_type = room_info.get('screenType')
+        live_source_type = room_info.get('liveSourceType')
+        stream_info_list = try_get(stream_data, lambda x: x['data'][0]['gameStreamInfoList'])
+        if not stream_info_list:
+            raise ExtractorError('Can not extract media info.')
```

Comment:
since u didn't answer, I assume this is not expected
In that case, why not just:
```suggestion
        stream_info_list = stream_data['data'][0]['gameStreamInfoList']
```

Reply:
Oh, sure, let me update